test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc

```diff
@@ -1088,6 +1088,87 @@ TEST_P(ExtProcIntegrationTest, GetAndFailStreamOnResponse) {
   verifyDownstreamResponse(*response, 500);
 }
 
+TEST_P(ExtProcIntegrationTest, OnlyRequestHeadersResetOnServerMessage) {
+  // Skip the header processing on response path.
```

Comment:
Nit: recommend adding:
"scoped_runtime_.mergeValues({{"envoy.reloadable_features.ext_proc_graceful_grpc_close", "false"}});"
here in case someone flip the default runtime config to true, then this test will fail.